deps/undici/src/lib/observable.js

```diff
@@ -0,0 +1,802 @@
+          isComplete = true;
+          for (const [deferredResolve] of deferred) {
+            deferredResolve({ value: undefined, done: true });
+          }
+          resolve({ value: undefined, done: true });
+        });
+      },
+      [Symbol.asyncIterator]() {
+        return this;
+      },
+    };
+  }
+}
+
+class Subscriber {
+  #active = true;
+  #abortController = new AbortController();
+  #signal;
+  #teardowns;
+
+  get signal() {
+    return this.#signal;
+  }
+
+  get isActive() {
+    return this.#active && !this.signal.aborted;
+  }
+
+  constructor(signal, _observer) {
+    this._observer = _observer;
+    const ownSignal = this.#abortController.signal;
+    this.#signal = signal ? AbortSignal.all([signal, ownSignal]) : ownSignal;
+  }
+
+  next(value) {
+    if (this.isActive) {
+      this._observer.next(value);
+    }
+  }
+
+  error(error) {
+    if (this.isActive) {
+      this.#active = false;
+      this._observer.error(error);
+      this.#abortController.abort();
+    }
+  }
+
+  complete() {
+    if (this.isActive) {
+      this.#active = false;
+      this._observer.complete();
+      this.#abortController.abort();
+    }
+  }
+
+  #teardownHandler = () => {
+    const teardowns = Array.from(this.#teardowns);
+    this.#teardowns = undefined;
+    let errors;
+    for (let i = teardowns.length - 1; i >= 0; i--) {
+      try {
+        teardowns[i]();
+      } catch (error) {
+        errors ??= [];
+        errors.push(error);
+      }
+    }
+    if (errors) {
+      // TODO: We need to figure out how to report multiple errors.
+      throw new Error(
+        `${errors.length} teardowns failed.\n ${errors.join('\n')}}`
+      );
+    }
+  };
+
+  addTeardown(teardown) {
+    if (this.isActive) {
+      if (!this.#teardowns) {
+        this.#teardowns = [];
+        this.#abortController.signal.addEventListener(
+          'abort',
+          this.#teardownHandler,
+          {
+            once: true,
+          }
+        );
+      }
+      this.#teardowns.push(teardown);
+    } else {
+      teardown();
+    }
+  }
+
+  removeTeardown(teardown) {
+    if (this.#teardowns) {
+      const index = this.#teardowns.indexOf(teardown);
+      if (index >= 0) {
+        this.#teardowns.splice(index, 1);
+        if (this.#teardowns.length === 0) {
+          this.#teardowns = undefined;
+          this.#abortController.signal.removeEventListener(
+            'abort',
+            this.#teardownHandler
+          );
+        }
+      }
+    }
+  }
+}
+
+function noop() {}
+
+/**
+ * Rethrows an error in its own call stack. This is done to prevent unwinding
+ * the stack in an asynchronous context where the error is uncatchable, possibly
+ * causing producer interference.
+ *
+ * In the browser, this would be reported via the `reportError` API there, to the
+ * global error handler and carry on. In Node, this will report and panic.
+ * @param {any} error The error to throw.
+ */
+function reportError(error) {
```

Comment:
This is used when the user doesn't provide an error handler to the `subscribe` call.

The behavior here matches what RxJS does. The browser will inherently be a little different, as `reportError` is an API there. The goal of this is commented on above... but the basics are: We don't want to rethrow the error immediately because it will unwind the stack and cause issues. We really want to report it so some unhandled error handler. One way of doing that in Node is to just throw it in a `setTimeout`.

Node may choose to be different in this regard though. Happy to debate the behavior, just as long as we're not synchronously rethrowing the error by default.